sources/packages/backend/apps/api/src/route-controllers/assessment/_tests_/e2e/assessment.students.controller.confirmAssessmentNOA.e2e-spec.ts

```diff
@@ -0,0 +1,140 @@
+  let appModule: TestingModule;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource, module } =
+      await createTestingAppModule();
+    app = nestApplication;
+    appModule = module;
+    db = createE2EDataSources(dataSource);
+  });
+
+  it("Should approve an assessment if requirements for approval are met.", async () => {
+    // Arrange
+
+    // Create the new student to be mocked as the authenticated one.
+    const student = await saveFakeStudent(db.dataSource);
+    // Mock user services to return the saved student.
+    await mockUserLoginInfo(appModule, student);
+    // MSFAA required for the NOA be returned.
+    const msfaaNumber = createFakeMSFAANumber(
+      { student },
+      {
+        msfaaState: MSFAAStates.Signed,
+      },
+    );
+    await db.msfaaNumber.save(msfaaNumber);
+    // Creates the applications to get the NOA.
+    const application = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      {
+        msfaaNumber,
+        student,
+      },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+      },
+    );
+    const oldAssessment = application.currentAssessment;
+
+    // Creates a new application's current assessment.
+    const newCurrentAssessment = await db.studentAssessment.save(
+      createFakeStudentAssessment(
+        {
+          auditUser: student.user,
+          application,
+          offering: application.currentAssessment.offering,
+        },
+        {
+          initialValue: {
+            triggerType: AssessmentTriggerType.RelatedApplicationChanged,
+          },
+        },
+      ),
+    );
+    application.currentAssessment = newCurrentAssessment;
+    await db.application.save(application);
+    // Create and save a new disbursement schedule for the new assessment.
+    const newAssessmentDisbursement = createFakeDisbursementSchedule({
+      studentAssessment: newCurrentAssessment,
+      msfaaNumber,
+      disbursementValues: [
+        createFakeDisbursementValue(
+          DisbursementValueType.CanadaLoan,
+          "CSLF",
+          1250,
+        ),
+        createFakeDisbursementValue(
+          DisbursementValueType.CanadaGrant,
+          "CSGP",
+          1500,
+        ),
+      ],
+    });
+    await db.disbursementSchedule.save(newAssessmentDisbursement);
+
+    const oldEndpoint = `/students/assessment/${oldAssessment.id}/confirm-assessment`;
+    const currentEndpoint = `/students/assessment/${application.currentAssessment.id}/confirm-assessment`;
+
+    const studentUserToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(oldEndpoint)
+      .auth(studentUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.UNPROCESSABLE_ENTITY);
+
+    await request(app.getHttpServer())
+      .patch(currentEndpoint)
+      .auth(studentUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK);
```

Comment:
@dheepak-aot @andrepestana-aot @sh16011993 @lewischen-aot in the past there was a direction to try to keep one endpoint call per E2E test trying to have a more well-defined target for the assertion.
Here there are two endpoint calls where the direction would be to have the "Arrange" part centralized in a private method and the two E2E scenarios targeting each endpoint call.
Thoughts? Is the direction still valid?

Reply:
IMO calling the endpoint with 2 different instances (2 different assessment ids) in the same test expands scope of assertion. I would prefer the scope of assertion well defined and have them as 2 different tests.

Reply:
Updated